packages/expo-updates/android/src/main/java/expo/modules/updates/loader/FileDownloader.java

```diff
@@ -118,6 +118,9 @@ public void onResponse(Call call, Response response) throws IOException {
           }
 
           try {
+            String expoProtocolVersion = response.header("expo-protocol-version", null);
+            configuration.setUsesLegacyManifest(expoProtocolVersion == null);
```

Comment:
IIRC we already pass the response into the `ManifestFactory` function -- could we just check the header and make the decision there? Would be nice to eliminate this key from the `UpdatesConfiguration` entirely and just rely on the response. (You might need to mock the response in the tests -- you can see an example of that in ManifestMetadataTest.java.)

Reply:
ah ok, that makes sense. Let me try.